template/src/features/counter/counterSlice.ts

```diff
@@ -29,18 +29,22 @@ export const slice = createSlice({
   },
 });
 
-// The function below is called a selector and allows us to select a value from
-// the state. Selectors can also be defined inline where they're used instead of
-// in the slice file. For example: `useSelector((state) => state.counter.value)`
-export const selectCount = (state: { counter: CounterState }) =>
-  state.counter.value;
-
 export const { increment, decrement, incrementByAmount } = slice.actions;
 
-export const incrementAsync = (): AppThunk => dispatch => {
+// The function below is called a thunk and allows us to perform async logic. It
+// can be dispatched like a regular action: `dispatch(incrementAsync(10))`. This
+// will call the thunk with the `dispatch` function as the first argument. Async
+// code can then be executed and other actions can be dispatched
+export const incrementAsync = (amount: number): AppThunk => dispatch => {
   setTimeout(() => {
-    dispatch(increment());
+    dispatch(incrementByAmount(amount));
   }, 1000);
 };
 
+// The function below is called a selector and allows us to select a value from
+// the state. Selectors can also be defined inline where they're used instead of
+// in the slice file. For example: `useSelector((state) => state.counter.value)`
+export const selectCount = (state: { counter: CounterState }) =>
```

Comment:
Should import the `RootState` type from `app/store.ts`, and use that here.  Let's also add `state: RootState` in the comment on line 46 as well.